homeassistant/components/fritz/image.py

```diff
@@ -0,0 +1,80 @@
+"""FRITZ image integration."""
+
+from __future__ import annotations
+
+from io import BytesIO
+import logging
+
+from homeassistant.components.image import ImageEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util, slugify
+
+from .common import AvmWrapper, FritzBoxBaseEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up guest WiFi QR code for device."""
+    avm_wrapper: AvmWrapper = hass.data[DOMAIN][entry.entry_id]
+
+    guest_wifi_info = await hass.async_add_executor_job(
+        avm_wrapper.fritz_guest_wifi.get_info
+    )
+
+    if not guest_wifi_info.get("NewEnable"):
+        return
+
+    async_add_entities(
+        [FritzGuestWifiQRImage(avm_wrapper, entry.title, guest_wifi_info["NewSSID"])]
+    )
+
+
+class FritzGuestWifiQRImage(FritzBoxBaseEntity, ImageEntity):
+    """Implementation of the FritzBox guest wifi QR code image entity."""
+
+    _attr_content_type = "image/png"
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    def __init__(
+        self, avm_wrapper: AvmWrapper, device_friendly_name: str, ssid: str
+    ) -> None:
+        """Initialize the image entity."""
+        self._attr_name = f"{device_friendly_name} {ssid} QR-Code"
+        self._attr_unique_id = slugify(f"{avm_wrapper.unique_id}-{ssid}-qr-code")
+        self._current_qr_bytes: bytes | None = None
+        super().__init__(avm_wrapper, device_friendly_name)
+        ImageEntity.__init__(self)
+
+    async def async_added_to_hass(self) -> None:
+        """Set the update time."""
+        self._attr_image_last_updated = dt_util.utcnow()
+
+    async def async_image(self) -> bytes:
+        """Return bytes of image."""
+        qr_stream: BytesIO = await self.hass.async_add_executor_job(
+            self._avm_wrapper.fritz_guest_wifi.get_wifi_qr_code, "png"
```

Comment:
Can we do SVG?

Reply:
I had it before, but with svg you got a problem with dark background color, because the "white" space between the black digits in the QR code is transparent (_will provide a screenshot today evening_)

Reply:
ok, let's keep it PNG.

Reply:
| png | svg |
| --- | --- |
| ![image](https://github.com/home-assistant/core/assets/35783820/f899eb58-7e54-487e-ab34-fa46c44c0339) | ![image](https://github.com/home-assistant/core/assets/35783820/385a0499-7fd7-4c32-8daa-9e0c340e1ff5) |

_qr code based on fake data_